src/diffusers/models/unet_1d_blocks.py

```diff
@@ -0,0 +1,315 @@
+        self.pad = kernel_1d.shape[0] // 2 - 1
+        self.register_buffer("kernel", kernel_1d)
+
+    def forward(self, x):
+        x = F.pad(x, (self.pad,) * 2, self.pad_mode)
+        weight = x.new_zeros([x.shape[1], x.shape[1], self.kernel.shape[0]])
+        indices = torch.arange(x.shape[1], device=x.device)
+        weight[indices, indices] = self.kernel.to(weight)
+        return F.conv1d(x, weight, stride=2)
+
+
+class Upsample1d(nn.Module):
+    def __init__(self, kernel="linear", pad_mode="reflect"):
+        super().__init__()
+        self.pad_mode = pad_mode
+        kernel_1d = torch.tensor(_kernels[kernel]) * 2
+        self.pad = kernel_1d.shape[0] // 2 - 1
+        self.register_buffer("kernel", kernel_1d)
+
+    def forward(self, x):
+        x = F.pad(x, ((self.pad + 1) // 2,) * 2, self.pad_mode)
+        weight = x.new_zeros([x.shape[1], x.shape[1], self.kernel.shape[0]])
+        indices = torch.arange(x.shape[1], device=x.device)
+        weight[indices, indices] = self.kernel.to(weight)
+        return F.conv_transpose1d(x, weight, stride=2, padding=self.pad * 2 + 1)
+
+
+class SelfAttention1d(nn.Module):
+    def __init__(self, c_in, n_head=1, dropout_rate=0.0):
+        super().__init__()
+        assert c_in % n_head == 0
+        self.norm = nn.GroupNorm(1, c_in)
+        self.n_head = n_head
+        self.qkv_proj = nn.Conv1d(c_in, c_in * 3, 1)
+        self.out_proj = nn.Conv1d(c_in, c_in, 1)
+        self.dropout = nn.Dropout(dropout_rate, inplace=True)
+
+    def forward(self, input):
+        n, c, s = input.shape
+        qkv = self.qkv_proj(self.norm(input))
+        qkv = qkv.view([n, self.n_head * 3, c // self.n_head, s]).transpose(2, 3)
+        q, k, v = qkv.chunk(3, dim=1)
+        scale = k.shape[3] ** -0.25
+        att = ((q * scale) @ (k.transpose(2, 3) * scale)).softmax(3)
+        y = (att @ v).transpose(2, 3).contiguous().view([n, c, s])
+        return input + self.dropout(self.out_proj(y))
+
+
+class SkipBlock(nn.Module):
+    def __init__(self, *main):
+        super().__init__()
+        self.main = nn.Sequential(*main)
+
+    def forward(self, input):
+        return torch.cat([self.main(input), input], dim=1)
+
+
+# Noise level (and other) conditioning
+class ResConvBlock(nn.Module):
+    def __init__(self, c_in, c_mid, c_out, is_last=False):
+        self.skip = nn.Identity if c_in == c_out else nn.Conv1d(c_in, c_out, 1, bias=False)
+        self.main = nn.Sequential(
+            [
+                nn.Conv1d(c_in, c_mid, 5, padding=2),
+                nn.GroupNorm(1, c_mid),
+                nn.GELU(),
+                nn.Conv1d(c_mid, c_out, 5, padding=2),
+                nn.GroupNorm(1, c_out) if not is_last else nn.Identity(),
+                nn.GELU() if not is_last else nn.Identity(),
+            ]
+        )
+
+    def forward(self, input):
+        return self.main(input) + self.skip(input)
+
+
+def get_down_block(down_block_type, c, c_prev):
```

Comment:
@natolambert - similar to unet_2d_blocks we'll have a new `unet_1d_blocks.py` file where you can define very customizable unet classes